http4s-client/src/main/scala/com/qvantel/jsonapi/client/http4s/JsonApiInstances.scala

```diff
@@ -0,0 +1,79 @@
+package com.qvantel.jsonapi.client.http4s
+
+import scala.language.higherKinds
+
+import scala.util.{Failure, Success, Try}
+import cats.Applicative
+import cats.data.EitherT
+import cats.effect._
+import cats.implicits._
+import org.http4s.{Charset, DecodeFailure, DecodeResult, EntityDecoder, EntityEncoder, MalformedMessageBodyFailure, MediaType, Message}
+import org.http4s.headers.`Content-Type`
+import spray.json._
+import spray.json.JsonParser.ParsingException
+
+import com.qvantel.jsonapi.{JsonApiReader, JsonApiWriter, readCollection, readOne, rawOne, rawCollection}
+
+final case class Include(include: Set[String])
+
+trait JsonApiInstances {
+  val mediaType: MediaType = MediaType.fromKey(("application", "vnd.api+json"))
+
+  implicit def jsonDecoder[F[_]: Effect, A: JsonApiReader](implicit include: Include): EntityDecoder[F, A] =
+    EntityDecoder.decodeBy[F, A](mediaType) { msg =>
+      msg.attemptAs[String].flatMap { x =>
+        val t: Either[DecodeFailure, A] = Try {
```

Comment:
Maybe just use `try { Right(readOne(…)) } catch { … }` to avoid allocating a `Try` just to immediately convert it to an `Either`?

Reply:
changed to try catch.